### PR TITLE
Add support for OAuth2 to OpenAPI3 importer

### LINKED ISSUE #2183: [FEATURE REQUEST] - Include OAuth2 settings when importing from an openapi3 file.
**Is your feature request related to a problem? Please describe.**
Currently when importing a workspace from an openapi3 file (with OAuth2 settings defined) the OAuth2 settings does not get applied to the requests.

I'd like to be able to generate and publish an Insomnia workspace along with our API documentation whenever our API specification updates.

**Describe alternatives you've considered**
To be able to generate the workspace with OAuth2 settings applied, I wrote a script that generates a workspace with the insomnia-importer, and then iterates every request, and adds the authentication settings to each request.

**Describe the solution you'd like**
I'd like `accessTokenUrl`,  `authorizationUrl`, and `scopes` to be parsed from the openapi file, and credentials(`username`, `password`, `clientId`, `clientSecret`) to be used as variables set in environment.

### PR DESCRIPTION
**Pull request by sebastianjonasson**
Add support for including OAuth2 settings when importing a workspace from an OpenAPI3 file.

Parses `authorizationUrl`, `accessToken`, and `scope` from OpenAPI specification file. Then adds templates/variables for user credentials (`password`, `username`, `client_id`, `client_secret`).

I will add tests if you want me to proceed. 

Closes #2183

### CODE REVIEW

**Review by develohpanda — APPROVED** [review]
Hi @sebastianjonasson, thanks for working on this! Apologies for the delay. 😞 

> I will add tests if you want me to proceed.

Code looks good! I will merge once these tests are in. 🤗

**Comment by sebastianjonasson** [comment]
@develohpanda - Made some updates and added test.

The pre-commit hook generated diffs in these files:
```
packages/openapi-2-kong/dist/index.js
packages/openapi-2-kong/dist/index.js.map
```
are the meant to be commited?

**Comment by develohpanda** [comment]
> are the meant to be commited?

No they shouldn't be. Your branch is out of date with the develop, so if you merge with develop that should fix up the `.gitignore` file and also the linting failures (run `npm run lint:fix`) CI is reporting.

File: packages/insomnia-importers/src/importers/openapi3.js [hunk 1]
```diff
@@ -546,6 +580,73 @@ function parseHttpAuth(scheme) {
...
+      return {
+        ...base,
+        clientSecret: '{{ oauth2ClientSecret }}',
+        username: '{{ oauth2Username }}',
+        password: '{{ oauth2Password }}',
+        accessTokenUrl: flow.tokenUrl,
+      };
+  }
+}
+
```
Comment (develohpanda) [inline packages/insomnia-importers/src/importers/openapi3.js:649]: I might be mistaken, but it doesn't look like this logic is being used anywhere. Should it still exist?

Comment (sebastianjonasson) [reply packages/insomnia-importers/src/importers/openapi3.js:649]: What makes you think that?

If the openapi spec has an oauth2 definition the `parseOAuth2` function will be called (row 346). `parseOAuth2` uses  `mapOAuth2GrantType` and `parseOAuth2Scopes`.

On row 381-394 we're extracting relevant variables to be set in the environment.

Comment (develohpanda) [reply packages/insomnia-importers/src/importers/openapi3.js:649]: Strange, I had done a search for these functions (in my browser) but it didn't return anything... 😞

**Review by develohpanda — APPROVED** [review]
Looks good now with the tests 👍

**Pull request merged**